pkg/parse/opts.go

```diff
@@ -89,9 +89,10 @@ type Options struct {
 // Parser represents a parser that can be pointed at and continuously parse a source.
 type Parser interface {
 	parseSource(ctx context.Context, state sourceState) ([]ast.FileObject, status.MultiError)
-	setSourceStatus(ctx context.Context, newStatus sourceStatus) error
-	setRenderingStatus(ctx context.Context, oldStatus, newStatus renderingStatus) error
-	SetSyncStatus(ctx context.Context, newStatus syncStatus) error
+	LoadStatus(ctx context.Context) (*ReconcilerStatus, error)
+	setSourceStatus(ctx context.Context, newStatus SourceStatus) error
+	setRenderingStatus(ctx context.Context, oldStatus, newStatus RenderingStatus) error
```

Comment:
It bugs me that some of these are public and some private. It doesn't make any sense to have private methods on an interface. But maybe I'll save that for a different PR.